src/poly/dcrt/poly.rs

```diff
@@ -154,6 +197,69 @@ impl Poly for DCRTPoly {
             })
             .collect()
     }
+
+    /// Convert the polynomial to a compact byte representation
+    /// The returned bytes vector is encoded as follows:
+    /// 1. The first byte contains the `max_byte_size`, namely the maximum byte size of any
+    ///    coefficient in the poly
+    /// 2. The next `ceil(n/8)` bytes contain a bit vector, where each bit indicates if the
+    ///    corresponding coefficient is negative and `n` is the ring dimension
+    /// 3. The remaining `n * max_byte_size` contain the coefficient values
+    fn to_compact_bytes(&self) -> Vec<u8> {
+        let modulus = self.ptr_poly.GetModulus();
+        let modulus_big: BigUint = BigUint::from_str(&modulus).unwrap();
+        let q_half = modulus_big.clone() / 2u8;
+
+        let coeffs = self.coeffs();
+        let ring_dimension = coeffs.len();
+
+        // Create a bit vector to store flags for negative coefficients
+        let bit_vector_byte_size = ring_dimension.div_ceil(8);
+        let mut bit_vector = vec![0u8; bit_vector_byte_size];
```

Comment:
instead of Vec<u8> we can use BitVec like this https://github.com/MachinaIO/diamond-io/blob/main/src/poly/dcrt/sampler/hash.rs#L96